Feature/105 Implement cast details screen show all

khaledeid1k opened the pull request.

nourelhodaahmed commented on design_system/src/main/java/com/moscow/cineverse/designSystem/component/MovieUI.kt:
```diff
@@ -1,6 +1,6 @@
 package com.moscow.cineverse.designSystem.component
 
-data class Movie(
+data class MovieUI(
```
i don't think this is right to add a data type in design system and do mapping from presentation.
you just use it in `movie poster card` and it's not a design system component you can just put it in presentation it doesn't use `material3` .
so just put it in presentation and let it take our `movie ui state` in presentation

nourelhodaahmed commented on design_system/src/main/java/com/moscow/cineverse/designSystem/component/home/CarouselAnimation.kt:
```diff
@@ -23,26 +23,25 @@ import androidx.compose.ui.composed
 import kotlin.math.absoluteValue
 
 @Composable
 fun CarouselAnimation(
```
also here, it should be in presentation module
you can add a common component package in presentation for all screens and put `CarouselAnimation` and `MoviePosterCard` in it

nourelhodaahmed commented on design_system/src/main/java/com/moscow/cineverse/designSystem/component/match/MatchCarouselAnimation.kt:
```diff
@@ -18,25 +18,23 @@ import androidx.compose.ui.composed
 import kotlin.math.absoluteValue
 
 @Composable
 fun MatchCarouselAnimation(
```
same as above

nourelhodaahmed commented on presentation/src/main/java/com/moscow/cineverse/screen/cast_details_show_all/ShowAllActorMoviesScreen.kt:
```diff
@@ -0,0 +1,173 @@
+import androidx.compose.material3.Button
+import androidx.compose.material3.CircularProgressIndicator
+import androidx.compose.material3.Scaffold
+import androidx.compose.material3.Text
```
you can't use` material3` in presentation

instead of `scaffold` just use box or column 
instead of `CircularProgressIndicator` we have one in design system
instead of `button and text` you can just put them in design system or use one we have 
this can be put in design system 
` Button(
                                onClick = interactionListener::onRefresh
                            ) {
                                Text(stringResource(R.string.retry))
                            }`

khaledeid1k committed:
Refactor: Standardize MovieUI model and its usage

This commit refactors the codebase to use a consistent `MovieUI` data class for representing movie information in the UI layer.

**Key Changes:**

*   **Renamed `Movie` to `MovieUI`:** The `Movie` data class in the `design_system` module has been renamed to `MovieUI` for clarity and to distinguish it from domain-layer models.
*   **Moved `MoviePosterCard`:** The `MoviePosterCard` composable has been moved from the `presentation` module to the `design_system` module, as it's a reusable UI component.
*   **Updated `ExploreScreenState`:**
    *   Replaced `MediaItemUi` with `MovieUI`.
    *   Changed `movies` to `moviesUI`.
    *   Adjusted `searchResult` to use `MovieUI`.
*   **Updated `ExploreViewModel`:**
    *   Modified `onGetMovieByGenreIdSuccess` and other methods to use `MovieUI` and `moviesUI`.
*   **Updated `ExploreMapper`:**
    *   The `toUi` extension functions for `Movie` and `Series` now return `MovieUI`.
    *   `posterPath` is now mapped to `posterUrl` in `MovieUI`.
*   **Updated Carousel Components:**
    *   `MatchCarouselAnimation` and `CarouselAnimation` in `design_system` now accept a list of `MovieUI` instead of `Movie`.
*   **Updated `ExploreScreen`:**
    *   The screen now correctly uses `MovieUI` when rendering `MoviePosterCard`.
*   **DI Update:** Added `ShowAllActorMoviesInteractionViewModel` to the `viewModels` Koin module.
*   **Minor UI Adjustment:** Reduced vertical padding in `MovieAppBar`.

nourelhodaahmed commented on presentation/src/main/java/com/moscow/cineverse/screen/cast_details_show_all/ShowAllActorMoviesState.kt:
```diff
@@ -0,0 +1,22 @@
+package com.moscow.cineverse.screen.cast_details_show_all
+
+import com.moscow.cineverse.designSystem.component.MovieUI
```
just use the `movie ui state` we have, you will find it in explore so just put it in another package as it's the model we will use in all screens

khaledeid1k committed:
feat: Implement "Show All Actor Movies" screen

This commit introduces the "Show All Actor Movies" screen and its related components:

- **ShowAllActorMoviesScreen.kt:** The main Composable function for the screen. It displays a list of movies an actor has appeared in, using a `LazyVerticalGrid`. It supports both grid and list view modes, switchable via a `ViewModeToggle`. It also handles loading and error states.
- **ShowAllActorMoviesContent.kt:** The content composable that handles the UI logic based on the current state (loading, error, or success).
- **ShowAllActorMoviesInteractionListener.kt:** Defines the interactions possible on this screen, such as refreshing data, changing view mode, clicking on a movie, and navigating back.
- **ShowAllActorMoviesInteractionViewModel.kt:** The ViewModel responsible for managing the state and handling interactions for the screen.
- **ShowAllActorMoviesState.kt:** Represents the UI state of the screen, including loading status, error messages, current view mode, and the list of movies.
- **ShowAllActorMoviesMapper.kt:** (Currently empty) Intended for mapping data models to UI models.

The screen features:
- A `MovieAppBar` for navigation and title display.
- A `LazyVerticalGrid` to display `MoviePosterCard` items.
- A `ViewModeToggle` to switch between grid and list layouts.
- Handling of loading and error states with appropriate UI elements.
- A preview function (`ShowAllActorMoviesPreview`) for development and testing.

khaledeid1k committed:
feat: Implement "Show All Actor Movies" screen

This commit introduces the "Show All Actor Movies" screen and its related components:

- **ShowAllActorMoviesScreen.kt:** The main Composable function for the screen. It displays a list of movies an actor has appeared in, using a `LazyVerticalGrid`. It supports both grid and list view modes, switchable via a `ViewModeToggle`. It also handles loading and error states.
- **ShowAllActorMoviesContent.kt:** The content composable that handles the UI logic based on the current state (loading, error, or success).
- **ShowAllActorMoviesInteractionListener.kt:** Defines the interactions possible on this screen, such as refreshing data, changing view mode, clicking on a movie, and navigating back.
- **ShowAllActorMoviesInteractionViewModel.kt:** The ViewModel responsible for managing the state and handling interactions for the screen.
- **ShowAllActorMoviesState.kt:** Represents the UI state of the screen, including loading status, error messages, current view mode, and the list of movies.
- **ShowAllActorMoviesMapper.kt:** (Currently empty) Intended for mapping data models to UI models.

The screen features:
- A `MovieAppBar` for navigation and title display.
- A `LazyVerticalGrid` to display `MoviePosterCard` items.
- A `ViewModeToggle` to switch between grid and list layouts.
- Handling of loading and error states with appropriate UI elements.
- A preview function (`ShowAllActorMoviesPreview`) for development and testing.

nourelhodaahmed commented on presentation/src/main/java/com/moscow/cineverse/screen/explore/ExploreMapper.kt:
```diff
@@ -4,30 +4,30 @@ import com.android.domain.model.Actor
 
 const val YYYY_MMM_DD = "yyyy, MMM dd"
 
-fun Movie.toUi(genresList: List<GenreUi>): MediaItemUi =
```
this function that you updated will give a lot of errors in base branch as we use `media item ui` in explore screen and there is another team working in refactor for the explore screen so we can't merge this 

just put another mapping function and don't update this

khaledeid1k committed:
Merge remote-tracking branch 'origin/feature/#4-cast-details_show_all' into feature/#4-cast-details_show_all

khaledeid1k committed:
Merge remote-tracking branch 'origin/feature/#4-cast-details_show_all' into feature/#4-cast-details_show_all

khaledeid1k committed:
Merge remote-tracking branch 'origin/feature/#4-cast-details_show_all' into feature/#4-cast-details_show_all

khaledeid1k committed:
Revert "Refactor: Standardize MovieUI model and its usage"

This reverts commit ef3a9c03

khaledeid1k committed:
Refactor: Extract `MoviePosterCard` and `MediaItemUi` to a common package

This commit moves `MoviePosterCard` and its data class `MediaItemUi` to a new package `com.moscow.cineverse.screen.component.movie_poster_card` for better organization and reusability across different screens.

**Key changes:**
- Created new package `com.moscow.cineverse.screen.component.movie_poster_card`.
- Moved `MoviePosterCard.kt` to the new package.
- Created `MediaItemUi.kt` in the new package and moved the `MediaItemUi` data class definition there from `ExploreScreenState.kt`.
- Updated import statements in `ShowAllActorMoviesScreen.kt` and `ExploreScreen.kt` to reflect the new location of `MoviePosterCard` and `MediaItemUi`.
- Renamed `moviesUI` to `movies` in `ShowAllActorMoviesState.kt` and `ShowAllActorMoviesScreen.kt` to align with the `MediaItemUi` naming.
- Removed vertical padding from `MovieAppBar` for a more consistent look.

khaledeid1k committed:
Refactor: Update ShowAllActorMoviesScreen to use MovieCircularProgressBar

The `ShowAllActorMoviesScreen` now utilizes the custom `MovieCircularProgressBar` component for displaying loading states, replacing the standard `CircularProgressIndicator`.

Gradient colors for the progress bar have been set to `Theme.colors.shade.primary` and `Theme.colors.shade.tertiary`.

nourelhodaahmed commented:
Good job, Just resolve the conflicts

khaledeid1k committed:
feat: Add `MovieScaffold`, `MovieText`, and `MovieButton` composables

This commit introduces three new reusable composables to the `design_system` module:

- **`MovieScaffold`**: A basic Scaffold wrapper that provides a consistent structure for screens, including a top app bar and floating action button slots. It also sets a default background color.
- **`MovieText`**: A customizable Text composable with default styling from the application's theme. It allows for overriding text color, alignment, max lines, overflow, and text style.
- **`MovieButton`**: A customizable Button composable with default styling for background, content color, disabled states, and shape. It accepts a text string and an onClick lambda.

Additionally, the `ShowAllActorMoviesScreen` in the `presentation` module has been updated to use these new composables (`MovieScaffold`, `MovieText`, `MovieButton`) instead of the standard Material3 components, promoting consistency and reusability.

khaledeid1k committed:
Merge branch 'feature/105-implement-cast-details-screen' into feature/105-implement-cast-details-screen_show_all
